### PR TITLE
ci: add github action to publish to dockerhub and github

### PR DESCRIPTION
**Pull request by thisisnithin**
Publishes to docker hub. Does not create a release (goreleaser doesn't work without semantic version)

Need to add the following to GitHub Secrets
- DOCKERHUB_USERNAME
- DOCKERHUB_TOKEN

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up QEMU
```
Comment (StarpTech) [inline .github/workflows/release.yaml]: Why do we need this?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        uses: docker/setup-qemu-action@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
```
Comment (StarpTech) [inline .github/workflows/release.yaml]: When we login to Dockerhub why do we login to ghcr?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (thisisnithin) [reply .github/workflows/release.yaml]: For multi-architecture images, I haven't specified the platforms in the build step. Which ones should I build for?

File: .github/workflows/release.yaml (hunk 2 above)
Comment (thisisnithin) [reply .github/workflows/release.yaml]: I was just pushing to both, I'll remove ghcr if all we want is dockerhub

File: .github/workflows/release.yaml (hunk 1 above)
Comment (StarpTech) [reply .github/workflows/release.yaml]: I'd say for all commons (darwin, linux, arm, windows).

File: .github/workflows/release.yaml (hunk 2 above)
Comment (StarpTech) [reply .github/workflows/release.yaml]: I'd say docker hub alone is fine.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (thisisnithin) [reply .github/workflows/release.yaml]: Done

File: .github/workflows/release.yaml (hunk 2 above)
Comment (thisisnithin) [reply .github/workflows/release.yaml]: Removed ghcr

**Comment by jensneuse** [comment]
> Publishes to docker hub. Does not create a release (goreleaser doesn't work without semantic version)
> 
> Need to add the following to GitHub Secrets
> 
> * DOCKERHUB_USERNAME
> * DOCKERHUB_TOKEN

You can use the following secrets:
DOCKER_PASSWORD,DOCKER_USERNAME

**Comment by StarpTech** [comment]
I'd run the workflow on this PR for testing.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (jensneuse) [reply .github/workflows/release.yaml]: as you now only build for linux, can be removed again, right?

**Review by jensneuse — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (thisisnithin) [reply .github/workflows/release.yaml]: Yes

Comment (thisisnithin) [reply .github/workflows/release.yaml]: Removed and tested

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
```
Comment (StarpTech) [inline .github/workflows/release.yaml:32]: We should run the tests before we release it to the hub.

**Review by StarpTech — CHANGES_REQUESTED** [review]
LGTM

File: .github/workflows/release.yaml (hunk 3 above)
Comment (thisisnithin) [reply .github/workflows/release.yaml:32]: Added test step

**Review by jensneuse — APPROVED** [review]

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**